nestjs/src/opportunities/dto/give-consent-dto.ts

```diff
@@ -0,0 +1,17 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsBoolean, IsOptional } from 'class-validator';
+
+export class GiveConsentDto {
+  constructor(consent: boolean, expires: number) {
+    this.consent = consent;
+    this.expires = expires;
+  }
+
+  @ApiProperty()
+  @IsBoolean()
+  public consent: boolean;
+
+  @ApiProperty()
+  @IsOptional()
+  public expires: number;
```

Comment:
it seems to me that either you need to remove `IsOptional` or `expires: number | undefined`